### PR TITLE
PHPUnit, The Bigger Picture

### LINKED ISSUE #3179: PHPUnit: The bigger picture
### Proposed title of article
PHPunit: The Bigger Picture

### Introduction paragraph (2-3 paragraphs):

This article is the first one in the series of articles that focus on `Unit Testing`, a well known test methodology. I will use `PHPUnit` as the main test tool for PHP Unit Testing. Once you get hold of the basics, getting started with PHPUnit is not that difficult.

Even a small software development project usually takes hours of hard work. During the course of development, a number of major and minor bugs and problems creep in the structure and code of the application. Developers usually try to work out the bugs during the process of development.

The problem is that there is no surefire way of ensuring that all the bugs have been worked out of the final code.

Another important aspect is the possibility of the introduction of further bugs when existing bugs are fixed. To address these problems, testing and QA are integral components of the software development process.

In this tutorial I will explain why we should write unit tests to cover the codes we are writing and the power of PHPUnit. At the end, we will write a simple test using PHPUnit.

### Key takeaways: 
1.What is Unit Testing?
2.Why do Unit Testing?
3.How to Install PHPUnit?
4.Basic Conventions to Write Unit Test Case

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by odongoalbert**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3179 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/phpunit-bigger-picture/index.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
+### Introduction
+
+Unit Tests in PHP or any other language are mainly written by developers as a good practice to help them identify and fix bugs in time, refactor code and serve as documentation for a unit of the program under test.
```
Comment (jumamiller) [inline content/articles/phpunit-bigger-picture/index.md:3]: Great work, start by removing all spaces beween the header and the first line of your paragraphs.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by jumamiller** [comment]
@odongoalbert,  I've noticed that there a function that you defined to take two parameters and at the same time taking an array. Kindly fix before we move forward with the review.

**Comment by odongoalbert** [comment]
@jumamiller, thanks for identifying that, fixing right away.

**Comment by odongoalbert** [comment]
@jumamiller, changes have been made. Thanks.

**Review by jumamiller — APPROVED** [review]
Great work @odongoalbert !

**Comment by hectorkambow** [comment]
Overall nice job gents very easy to follow along

**Pull request merged**